### ISSUE TITLE
RedTrainer not randomizing attacks

### ISSUE THREAD

**Issue by oppalla** [yzsvdu/RedTrainer#3]
I've been trying to up some Onix and Dratini while I work but, it uses only the first move, and as soon as the PP ends, it stops doing anything.
Is there anyway to use random attacks or set one attack per Pokemon?
Is it too difficult to restore Pokemon Healths and PP on Pokecenter after a while?

Thanks for the bot

**Comment by yzsvdu** [comment]
Hello, when you say you are trying to "up" does that mean level up? When the PP ends for the first battle move it should use the second combat move and so on depending on the priority selected in the control panel. Is it not doing so?

**Comment by yzsvdu** [comment]
As for the pokecenter reset feature, I have a working version that does payday forever but its running on a fixed path which i won't release for now because i think its easy to get caught even though I have ran it for >24 hours without stop. In the future I plan on adding a record movement feature so that you can record your movement once and then it will create different patterns but still bring you to the same end result.

**Comment by oppalla** [comment]
yes, when the PP finish it goes to the second attack that is set in the bot window.
there is a way to select next pokemon as well if the first one faint?
and someway to realize when some attack is disabled by psychic pokemons, to use the secon one from the list?
thans

**Comment by LanggXuee** [comment]
> As for the pokecenter reset feature, I have a working version that does payday forever but its running on a fixed path which i won't release for now because i think its easy to get caught even though I have ran it for >24 hours without stop. In the future I plan on adding a record movement feature so that you can record your movement once and then it will create different patterns but still bring you to the same end result.

Hi, i do not know how to make a "pull request" so i will ask here, any advancements about your works? i'm really interested in this bot, i tested it and still run it everyday, but the movement are always the same, a lot of options are greyed out (not accessible) too, if i run it i know it will get suspected really quick because it always do 3 step left and right (or up and down etc), the bot also spam key a lot and go fast, it's good and bad, if you see a player doing the exact same little 3 movement over and over and always landing attacks extremly fast in all the fights, then starting to move INSTANTLY after a fight, peoples should know it's a bot, maybe there's a way to avoid all of this by adding random timer with random few "stop" and movements, sometimes it will take a little break and slow down, waiting few random seconds in fight before doing an attack, same for walking with even maybe sometimes it add more steps, like instead of doing 3 steps all the time, it will do 3 3 3 3 3 3 3 3 3 then 4 then 3 3 3 4 33 4, hope you get what i'm trying to say, the more human lazy random movements it can look, the better and less suspect it will be

If you do an auto center/path back to the farm spot, maybe adding multiples paths to make it look humans too instead of doing the exact same route and step over and over again without any failures


If i remember i saw my friends showing me screenshot of others ppl creating paid bot for this games, they have ways to launch multiples games on the same pc by bypassing the restriction, they run a lot of bots making millions everyday and they said maximum payday time for a bot per day is around 6 hours max, otherwise it will be too suspect and make too much money which will flag the account and put a lot of captcha on screen (i had captcha every 10-20 fights, not even joking, literally every few minutes = one captcha and i wasn't botting, i used to play on 3 devices IRL with my own hands, farming money by snatching amulet coin since it's by hordes so it's slow and doesn't need me to press the keyboards all the time on all 3 devices, i was making 600K an hour and shiny every 2-3 days, when i was giving amulets coins on my main account after being done with the farm, my account got flagged since it instantly received a lot of items worth a lot of moneys, and the account by itself even without my others accounts was getting a lot of money since i was farming 2-5 hours a day

Shiny hunting was 12 to 24 hours per day i totally forgot but it was a big number so this one is pretty safe, i do not know if killing hordes to get XP while doing 12 hours of farm will make it flag more or not, since you interact in the fights instead of running away and these mons can be sold when they are level 100 so i would avoid doing xp for 12-24 hours straight

BIG thing is : if accounts get flagged and/or reported, a GM can come and look at you, they will check the data abouts your account, they have access to memory and shit, they said it publicly on their websites and i guess this is why most bots simulate a lot of devices to try to bypass this and make it look like we are pressing everything in their eyes, but if an human manually check everything , he could find different infos and also spec you in games, this is also why it's important to have a very randomized bot so you don't look like a dumb ass doing the same steps over and over, i saw a bot from 2013 that had options to automatically respond to peoples dm to try to look like an human and get away, i think this is important too, if ppl talk to you they can test if you're a bot by that, whenever i use YOUR bot, i sometimes talk to random peoples and i stop the bot every few 20-50 minutes to manually farm a bit then i start the bot again, i switch farm places too and press multiples key, sometimes i open the shop, the dex, the trainer card, my pokemons info etc, i just try the maximum to look like an humain doing his normal adventure



Long boring text but everyone can learn from it, let's avoid getting flagged, thanks for reading i appreciate your work!

**Comment by yzsvdu** [comment]
I've been a little busy lately and considered open sourcing the entire project but I think that would make it easier to get banned. When I am more free I can work on more features.

**Comment by LanggXuee** [comment]
> I've been a little busy lately and considered open sourcing the entire project but I think that would make it easier to get banned. When I am more free I can work on more features.

yes sadly, open source mean we can trust your program (and i also believe 99% of all others bots are stealing datas, a lot of accounts are sold i won't be surprised if the bots owners who sell it just straight up steal names and passwords from their users, they just have to set up something to see if users are inactives and none of their bots are open sources, all are easily flagged by microsoft defender and virus total, i cannot trust anyone except your bot

**Comment by LanggXuee** [comment]
(but this does mean pokemmo dev can read it and easily patch the bot)

**Comment by LanggXuee** [comment]
> I've been a little busy lately and considered open sourcing the entire project but I think that would make it easier to get banned. When I am more free I can work on more features.

Hello, sorry for all my comments but i'm extremly curious and i really plan to be involved into botting, chinese bot farms use distance access (like for exemple here in europe we could use team viewers for it), on one pc they run a game, on another one a bot program, the bot will read the game on the screen by that "team viewer" distant access app and you can play the game at distance since the applications let you do it, games cannot detect and counter this, they cannot detect a bot program since it's not on the pc that run the game, you think we can use this on pokemmo? maybe a VM with the bot on, real pc run the game and we just install team viewer on the VM, connect it to our normal pc running the game so the bot on the VM can read the game and press keys, they will not detect anything from memories/data since the bot is "not here"

**Comment by yzsvdu** [comment]
> > I've been a little busy lately and considered open sourcing the entire project but I think that would make it easier to get banned. When I am more free I can work on more features.
> 
> 
> 
> Hello, sorry for all my comments but i'm extremly curious and i really plan to be involved into botting, chinese bot farms use distance access (like for exemple here in europe we could use team viewers for it), on one pc they run a game, on another one a bot program, the bot will read the game on the screen by that "team viewer" distant access app and you can play the game at distance since the applications let you do it, games cannot detect and counter this, they cannot detect a bot program since it's not on the pc that run the game, you think we can use this on pokemmo? maybe a VM with the bot on, real pc run the game and we just install team viewer on the VM, connect it to our normal pc running the game so the bot on the VM can read the game and press keys, they will not detect anything from memories/data since the bot is "not here" 

I'm sure something like that would work but might be inefficient because you need to read the data on the screen and transfer it over to the vm pc or have the vm pc do OCR.

**Comment by LanggXuee** [comment]
> > > I've been a little busy lately and considered open sourcing the entire project but I think that would make it easier to get banned. When I am more free I can work on more features.
> > 
> > 
> > Hello, sorry for all my comments but i'm extremly curious and i really plan to be involved into botting, chinese bot farms use distance access (like for exemple here in europe we could use team viewers for it), on one pc they run a game, on another one a bot program, the bot will read the game on the screen by that "team viewer" distant access app and you can play the game at distance since the applications let you do it, games cannot detect and counter this, they cannot detect a bot program since it's not on the pc that run the game, you think we can use this on pokemmo? maybe a VM with the bot on, real pc run the game and we just install team viewer on the VM, connect it to our normal pc running the game so the bot on the VM can read the game and press keys, they will not detect anything from memories/data since the bot is "not here"
> 
> I'm sure something like that would work but might be inefficient because you need to read the data on the screen and transfer it over to the vm pc or have the vm pc do OCR.

i don't have any knowledge on the subject but the if the bot is on a VM, on that VM you have team viewers, meaning the bot can see the game and control the game (moving characters and everything) thru that team viewers, it can't work? the bot is reading the screen no? it could read the screen on that team viewer thing?

**Comment by yzsvdu** [comment]
> > > > I've been a little busy lately and considered open sourcing the entire project but I think that would make it easier to get banned. When I am more free I can work on more features.
> > > 
> > > 
> > > Hello, sorry for all my comments but i'm extremly curious and i really plan to be involved into botting, chinese bot farms use distance access (like for exemple here in europe we could use team viewers for it), on one pc they run a game, on another one a bot program, the bot will read the game on the screen by that "team viewer" distant access app and you can play the game at distance since the applications let you do it, games cannot detect and counter this, they cannot detect a bot program since it's not on the pc that run the game, you think we can use this on pokemmo? maybe a VM with the bot on, real pc run the game and we just install team viewer on the VM, connect it to our normal pc running the game so the bot on the VM can read the game and press keys, they will not detect anything from memories/data since the bot is "not here"
> > 
> > 
> > I'm sure something like that would work but might be inefficient because you need to read the data on the screen and transfer it over to the vm pc or have the vm pc do OCR.
> 
> i don't have any knowledge on the subject but the if the bot is on a VM, on that VM you have team viewers, meaning the bot can see the game and control the game (moving characters and everything) thru that team viewers, it can't work? the bot is reading the screen no? it could read the screen on that team viewer thing?

If you are talking about this bot, it does not do character recognition / read from the screen. It reads values directly from the computer memory. For example, when you get an encounter, the game loads the encountered Pokemon into computer memory like their HP, Stats, Names etc and the bot reads from the place where the game stores it and makes decisions based on that. If you want to read more on this its called Java Reflection.

**Comment by LanggXuee** [comment]
> > > > > I've been a little busy lately and considered open sourcing the entire project but I think that would make it easier to get banned. When I am more free I can work on more features.
> > > > 
> > > > 
> > > > Hello, sorry for all my comments but i'm extremly curious and i really plan to be involved into botting, chinese bot farms use distance access (like for exemple here in europe we could use team viewers for it), on one pc they run a game, on another one a bot program, the bot will read the game on the screen by that "team viewer" distant access app and you can play the game at distance since the applications let you do it, games cannot detect and counter this, they cannot detect a bot program since it's not on the pc that run the game, you think we can use this on pokemmo? maybe a VM with the bot on, real pc run the game and we just install team viewer on the VM, connect it to our normal pc running the game so the bot on the VM can read the game and press keys, they will not detect anything from memories/data since the bot is "not here"
> > > 
> > > 
> > > I'm sure something like that would work but might be inefficient because you need to read the data on the screen and transfer it over to the vm pc or have the vm pc do OCR.
> > 
> > 
> > i don't have any knowledge on the subject but the if the bot is on a VM, on that VM you have team viewers, meaning the bot can see the game and control the game (moving characters and everything) thru that team viewers, it can't work? the bot is reading the screen no? it could read the screen on that team viewer thing?
> 
> If you are talking about this bot, it does not do character recognition / read from the screen. It reads values directly from the computer memory. For example, when you get an encounter, the game loads the encountered Pokemon into computer memory like their HP, Stats, Names etc and the bot reads from the place where the game stores it and makes decisions based on that. If you want to read more on this its called Java Reflection.

interesting, anyway as long as we can avoid bans in general i'm all good, keep the good work up bro! thanks a lot

**Comment by yzsvdu** [comment]
> > > > > > I've been a little busy lately and considered open sourcing the entire project but I think that would make it easier to get banned. When I am more free I can work on more features.
> > > > > 
> > > > > 
> > > > > Hello, sorry for all my comments but i'm extremly curious and i really plan to be involved into botting, chinese bot farms use distance access (like for exemple here in europe we could use team viewers for it), on one pc they run a game, on another one a bot program, the bot will read the game on the screen by that "team viewer" distant access app and you can play the game at distance since the applications let you do it, games cannot detect and counter this, they cannot detect a bot program since it's not on the pc that run the game, you think we can use this on pokemmo? maybe a VM with the bot on, real pc run the game and we just install team viewer on the VM, connect it to our normal pc running the game so the bot on the VM can read the game and press keys, they will not detect anything from memories/data since the bot is "not here"
> > > > 
> > > > 
> > > > I'm sure something like that would work but might be inefficient because you need to read the data on the screen and transfer it over to the vm pc or have the vm pc do OCR.
> > > 
> > > 
> > > i don't have any knowledge on the subject but the if the bot is on a VM, on that VM you have team viewers, meaning the bot can see the game and control the game (moving characters and everything) thru that team viewers, it can't work? the bot is reading the screen no? it could read the screen on that team viewer thing?
> > 
> > 
> > If you are talking about this bot, it does not do character recognition / read from the screen. It reads values directly from the computer memory. For example, when you get an encounter, the game loads the encountered Pokemon into computer memory like their HP, Stats, Names etc and the bot reads from the place where the game stores it and makes decisions based on that. If you want to read more on this its called Java Reflection.
> 
> interesting, anyway as long as we can avoid bans in general i'm all good, keep the good work up bro! thanks a lot

Yeah, the reason why the bot feels like it does things instantly is because it can directly read from memory and interact with buttons without using keyboard and mouse.  I could set a delay. The halloween event also breaks the bot temporarily, if there is enough users I can work on a fix soon https://github.com/yzsvdu/RedTrainer/discussions/8

**Comment by oppalla** [comment]
Nice discussion btw
